crates/ruff_linter/src/checkers/tokens.rs

```diff
@@ -44,22 +43,26 @@ pub(crate) fn check_tokens(
         Rule::BlankLinesBeforeNestedDefinition,
     ]) {
         BlankLinesChecker::new(locator, stylist, settings, source_type, cell_offsets)
-            .check_lines(tokens, &mut diagnostics);
+            .check_lines(program.tokens(), &mut diagnostics);
```

Comment:
Nit: You could assigned `tokens` and `comment_ranges` to local variables to reduce the diff size. 

```rust

			let tokens = program.tokens();
			let comment_ranges = program.comment_ranges();

			BlankLinesChecker::new(locator, stylist, settings, source_type, cell_offsets)
            .check_lines(tokens, &mut diagnostics);
```